lib/Doctrine/Common/Annotations/CachedReader.php

```diff
@@ -51,19 +53,37 @@ final class CachedReader implements Reader
     private $loadedAnnotations = [];
 
     /**
-     * Constructor.
+     * @deprecated use fromPsr16Cache instead
      *
-     * @param Reader $reader
-     * @param Cache  $cache
-     * @param bool   $debug
+     * @param Cache|CacheInterface $cache
+     * @param bool                 $debug
      */
-    public function __construct(Reader $reader, Cache $cache, $debug = false)
+    public function __construct(Reader $reader, $cache, $debug = false)
     {
+        if ($cache instanceof Cache) {
+            @trigger_error(sprintf(
+                'Passing a %s instance to %s is deprecated since doctrine/annotations 1.9 and will not be possible anymore in 2.0. Please pass a %s instance',
+                Cache::class,
+                __METHOD__,
+                CacheInterface::class
+            ), E_USER_DEPRECATED);
+            $cache = new SimpleCacheAdapter($cache);
+        }
         $this->delegate = $reader;
-        $this->cache = $cache;
+        $this->cache = (static function (CacheInterface $cache): CacheInterface {
+            return $cache;
+        })($cache);
         $this->debug = (boolean) $debug;
     }
 
+    public static function fromPsr16Cache(
```

Comment:
should it be `from` or `for` ?

Reply:
The implication is "Build a cached reader from a PSR 16 cache (and a wrapped cache)". We build it for the wrapped reader, from a cache. At least that's my reasoning, it would be great to have a native speaker chime in here.